### PR TITLE
[Core] Add multi-step support to LLMEngine

### PR DESCRIPTION
**Pull request by alexm-redhat**
This PR adds multi-step support to the LLM Engine class. The new multi-step methods that were originally added to async_llm_engine were moved to llm_engine, since async_llm_engine inherits from llm_engine.

### CODE REVIEW

**Comment by alexm-neuralmagic** [comment]
@robertgshaw2-neuralmagic @SolitaryThinker @comaniac

File: benchmarks/benchmark_throughput.py [hunk 1]
```diff
@@ -106,6 +106,8 @@ def run_vllm(
         max_num_batched_tokens=max_num_batched_tokens,
         distributed_executor_backend=distributed_executor_backend,
         load_format=load_format,
+        num_scheduler_steps=8,
+        use_v2_block_manager=True,
```
Comment (comaniac) [inline benchmarks/benchmark_throughput.py:110]: Make it configurable.

File: examples/offline_inference.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 sampling_params = SamplingParams(temperature=0.8, top_p=0.95)
 
 # Create an LLM.
-llm = LLM(model="facebook/opt-125m")
+llm = LLM(model="facebook/opt-125m", tensor_parallel_size=2)
```
Comment (comaniac) [inline examples/offline_inference.py:14]: Revert this

File: tests/multi_step/test_correctness.py [hunk 3]
```diff
@@ -48,9 +49,10 @@ async def completions_with_server_args(prompts: List[str], model_name: str,
 @pytest.mark.parametrize("num_scheduler_steps", NUM_SCHEDULER_STEPS)
 @pytest.mark.parametrize("num_prompts", NUM_PROMPTS)
 @pytest.mark.asyncio
-async def test_multi_step(example_prompts, model: str, tp_size: int,
-                          pp_size: int, eager_mode: int,
-                          num_scheduler_steps: int, num_prompts: int):
+async def test_multi_step_async_llm(example_prompts, model: str, tp_size: int,
+                                    pp_size: int, eager_mode: int,
+                                    num_scheduler_steps: int,
+                                    num_prompts: int):
```
Comment (comaniac) [inline tests/multi_step/test_correctness.py:55]: Revert this as unrelated

**Review by comaniac — APPROVED** [review]
LGTM. cc @SolitaryThinker

File: benchmarks/benchmark_throughput.py (hunk 1 above)
Comment (alexm-neuralmagic) [reply benchmarks/benchmark_throughput.py:110]: Good catch, added as arg params

File: examples/offline_inference.py (hunk 2 above)
Comment (alexm-neuralmagic) [reply examples/offline_inference.py:14]: reverted

File: tests/multi_step/test_correctness.py (hunk 3 above)
Comment (alexm-neuralmagic) [reply tests/multi_step/test_correctness.py:55]: reverted

**Review by mgoin — APPROVED** [review]
Excellent, great for offline throughput

**Review by SolitaryThinker — APPROVED** [review]
lgmt thanks @alexm-neuralmagic!

**Pull request merged**